pkg/kv/batch.go

```diff
@@ -140,24 +145,38 @@ func prev(ba roachpb.BatchRequest, k roachpb.RKey) (roachpb.RKey, error) {
 		if err != nil {
 			return nil, err
 		}
+		// FIXINPR(tschottdorf,nvanbenschoten): yikes, this is totally paged out.
```

Comment:
Range-local keys exist in the "gaps" between regular keys. `keys.Addr` finds the lower bound: the greatest regular key that is less than or equal to the input, and AddrUpperBound finds the smallest regular key that is greater than or equal to the input. 

Here, we're trying to expand `[StartKey, EndKey)` to find the keys that form the tightest (potential) range boundaries around the given keys. So we must use Addr on StartKey and AddrUpperBound on EndKey to avoid leaving EndKey outside the new bounds.

Reply:
Thanks for the reminder, added misc commentary.

Reply:
Done.